gui/src/main.rs

```diff
@@ -97,195 +115,237 @@ impl Gui {
-                ui.label("URL:");
-                ui.text_edit_singleline(&mut self.config.url);
-                if ui.button("Submit").clicked() {
-                    let body = if self.config.selected_http_method != HttpMethod::GET {
-                        Some(
-                            serde_json::from_str(&self.config.body_str)
-                                .expect("Body is invalid json"),
-                        )
-                    } else {
-                        None
-                    };
-                    let submitted_headers = self
-                        .config
-                        .headers
-                        .borrow_mut()
-                        .iter()
-                        .filter(|h| h.0 == true)
-                        .map(|h| (h.1.to_owned(), h.2.to_owned()))
-                        .collect();
-                    let request = HttpRequest {
-                        id: Uuid::new_v4(),
-                        name: None,
-                        headers: Some(Gui::remove_duplicate_headers(submitted_headers)),
-                        body,
-                        method: self.config.selected_http_method.clone(),
-                        url: self.config.url.clone(),
-                    };
-
-                    let response = Gui::spawn_submit(self, request);
-                    if response.is_ok() {
-                        self.config.response = Some(response.unwrap());
-                    }
-                }
+                    ui.menu_button("Import", |ui| {
+                        if ui.button("Collection").clicked() {
+                            if let Ok(mut config) = self.config.try_write() {
+                                (*config).import_window_open = true;
+                                (*config).import_mode = ImportMode::COLLECTION;
+                            }
+                        };
+                        if ui.button("Environment").clicked() {
+                            if let Ok(mut config) = self.config.try_write() {
+                                (*config).import_window_open = true;
+                                (*config).import_mode = ImportMode::ENVIRONMENT;
+                            }
+                        };
+                    });
+                    ui.menu_button("Export", |ui| {
+                        if ui.button("Collection").clicked() {
+                            ui.close_menu();
+                        };
+                        if ui.button("Environment").clicked() {
+                            ui.close_menu();
+                        };
+                    });
+                });
             });
-            ui.horizontal(|ui| {
-                if ui.button("Params").clicked() {
-                    self.config.request_window_mode = RequestWindowMode::PARAMS;
+        });
+        SidePanel::left("nav_panel").show(ctx, |ui| {
+            if let Ok(mut config) = self.config.try_write() {
```

Comment:
Any bits of ui that go and update the `config` property are wrapped in this if let statement for acquiring a write lock.